client/src/features/visualizer-vivagraph/hooks/useFeed.ts

```diff
@@ -121,33 +261,47 @@ export const useFeed = (network: string) => {
                 theta: 0.8,
             });
 
-            graphics.current.setNodeProgram(buildNodeShader());
-
-            const events = Viva.Graph.webglInputEvents(graphics.current, graph.current);
+            graphContext.graphics.current.setNodeProgram(buildNodeShader());
+            const events = Viva.Graph.webglInputEvents(graphContext.graphics.current, graphContext.graph.current);
 
+            events.click((node) => selectNode(node));
             events.dblClick((node) => {
                 window.open(`${window.location.origin}/${network}/block/${node.id}`, "_blank");
             });
 
-            renderer.current = Viva.Graph.View.renderer<INodeData, unknown>(graph.current, {
-                container: graphElement.current,
-                graphics: graphics.current,
+            graphContext.renderer.current = Viva.Graph.View.renderer<INodeData, unknown>(graphContext.graph.current, {
+                container: graphContext.graphElement.current,
+                graphics: graphContext.graphics.current,
                 layout,
                 renderLinks: true,
             });
 
-            renderer.current.run();
+            graphContext.renderer.current.run();
 
-            graphics.current.scale(1, { x: graphElement.current.clientWidth / 2, y: graphElement.current.clientHeight / 2 });
+            graphContext.graphics.current.scale(1, {
+                x: graphContext.graphElement.current.clientWidth / 2,
+                y: graphContext.graphElement.current.clientHeight / 2,
+            });
 
             for (let i = 0; i < 12; i++) {
-                renderer.current.zoomOut();
+                graphContext.renderer.current.zoomOut();
             }
+
+            graphContext.setIsVivaReady(true);
```

Comment:
since the type is null | DIspatch 
```suggestion
            graphContext.setIsVivaReady?.(true);
```